### DISCUSSION TITLE
How to include timestamp in a log event?

### DISCUSSION THREAD

**Discussion by patrickdung** [vectordotdev/vector#16762] [Q&A] [answered]
Currently the timestamp is included inside the log structure:
log: { attribute1: data1, timestamp: datetime1}

How to make vector to send to Clickhouse as:
log: { attribute1: data1}, timestamp: datetime1

Fluent bit can append the time stamp next to the log message
How to perform the same thing with Vector?
https://clickhouse.com/blog/nginx-logs-to-clickhouse-fluent-bit

Thank you in advance.

**Comment by jszwedko** [comment, marked as answer]
Hi @patrickdung ,

You can use a `remap` transform with a VRL program like:

```
.log.attribute.timestamp = del(.log.timestamp)
```
